elements/pfe-navigation/src/pfe-navigation.js

```diff
@@ -0,0 +1,322 @@
+  };
+}
+
+class PfeNavigation extends PFElement {
+  static get tag() {
+    return "pfe-navigation";
+  }
+
+  get templateUrl() {
+    return "pfe-navigation.html";
+  }
+
+  get styleUrl() {
+    return "pfe-navigation.scss";
+  }
+
+  get schemaUrl() {
+    return "pfe-navigation.json";
+  }
+
+  get overlay() {
+    return !this._overlay.hasAttribute("hidden");
+  }
+
+  set overlay(state) {
+    if (state) {
+      // Add the overlay to the page
+      this._overlay.removeAttribute("hidden");
+      // This prevents background scroll while nav is open
+      document.body.style.overflow = "hidden";
+    } else {
+      // Remove the overlay from the page
+      this._overlay.setAttribute("hidden", "");
+      // Allow background to scroll again
+      document.body.style.overflow = "auto";
+
+    }
+  }
+
+  constructor() {
+    PFElement._debugLog = true;
+    super(PfeNavigation);
+
+    // Attach functions for use below
+    this._init = this._init.bind(this);
+    this._setVisibility = this._setVisibility.bind(this);
+
+    // -- handlers
+    this._keydownHandler = this._keydownHandler.bind(this);
+    this._resizeHandler = this._resizeHandler.bind(this);
+    this._stickyHandler = this._stickyHandler.bind(this);
+    this._outsideListener = this._outsideListener.bind(this);
+    this._observer = new MutationObserver(this._init);
+
+    // Capture shadow elements
+    this._overlay = this.shadowRoot.querySelector(".pfe-navigation__overlay");
+    this._menuItem = this.shadowRoot.querySelector("pfe-navigation-item");
+
+    // Initialize active navigation item to empty array
+    this._activeNavigationItems = [];
+    this.overlay = false;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    // If this element contains light DOM, initialize it
+    if (this.children.length) {
+      // If only one value exists in the array, it starts at that size and goes up
+      this.breakpoints = {
+        main: [0, 1199], // visible from 0 - 1200px
+        search: [768],   // visible from 768px +
+        "mobile-search": [0, 767],
+        language: [768],
+        "mobile-language": [0, 767],
+        login: [768],
+        "mobile-login": [0, 767]
+      };
+
+      // Kick off the initialization of the light DOM elements
+      this._init();
+
+      // Listen for tab events
+      this.addEventListener("keydown", this._keydownHandler);
+
+      // Watch for screen resizing
+      window.addEventListener("resize", this._resizeHandler);
+    } else {
+      console.error("This component does not have any light DOM children.  Please check documentation for requirements.");
+    }
+    
+    PFElement._debugLog = false;
+  }
+
+  disconnectedCallback() {
+    // Remove the scroll, resize, and outside click event listeners
+    window.removeEventListener("resize", this._resizeHandler);
+    window.removeEventListener("scroll", this._stickyHandler);
+    document.removeEventListener("click", this._outsideListener);
```

Comment:
The listener for click is added in a condition so we should only remove the event listener if the event was listened for in the first place.

Reply:
Updated